Port twisted.protocols.finger to Python 3

rodrigc opened the pull request:
http://twistedmatrix.com/trac/ticket/8230

rodrigc committed:
Switch to byte strings to fix tests on Python 3.

rodrigc committed:
Allow finger modules to be installed with Python 3

rodrigc committed:
Rename variable to appease twistedchecker

rodrigc committed:
Make sure we return a list, instead of generator, to fix example on Python 3

rodrigc committed:
Make sure we return a list, instead of generator, to fix example on Python 3

rodrigc committed:
Fix example on Python 3

rodrigc committed:
Add topfile

rodrigc commented:
@leth thanks for your previous work on Twisted.  Can you code review this on GitHub?

leth commented:
The diff looks reasonable :) Just a couple of minor subjective comments!

I note there's no decoding going on; from an API design point of view, does it make sense for this API to be returning bytes?

leth commented on docs/core/howto/tutorial/listings/finger/fingerXRclient.py:
```diff
@@ -1,7 +1,14 @@
+except ImportError:
+    # Python 2
+    import xmlrpclib
+    Server = xmlrpclib.Server
```
You could use "from" imports here

rodrigc replied:
OK

leth commented:
There are a couple of places with docstrings which say "strings" e.g. `IFingerSetterFactory`, which will be at odds with python3 if things are bytes.

rodrigc committed:
Change docstring from string to bytes, and use bytes in a few places.

This fixes some examples on Python 3.

rodrigc committed:
Fix some bytes/unicode issues in examples to fix Python 3

rodrigc committed:
Use imports

rodrigc committed:
Merge branch 'trunk' into 8230-rodrigc-finger

rodrigc commented:
@leth the finger protocol itself is quite old, and deals with bytes.  The code in Twisted is also quite old, since it was one of the first protocols written.  Now, not a lot of people are going to be implementing finger servers these days.  However, there is a nice Twisted tutorial based on this protocol, and it would be really nice if the examples worked more or less out of the box on Python 3,
to give new users a nice experience, instead of running into a lot of errors.

Based on your comments, I took a pass through the tutorial and changed the docstring for things like `IFingerSetterFactory` to bytes.  I also fixed up some of the examples in the finger tutorial.  Before, none of the examples worked on Python 3, but now quite a lot of them do.

I didn't get to all of them, but if it is OK with you, I'd like to commit this stuff to begin with,
and either I or other people can fix up the other examples over time.

Thanks!

leth commented:
Sounds reasonable!

On 29 January 2017 22:48:07 GMT+00:00, Craig Rodrigues <notifications@github.com> wrote:
>@leth the finger protocol itself is quite old, and deals with bytes. 
>The code in Twisted is also quite old, since it was one of the first
>protocols written.  Now, not a lot of people are going to be
>implementing finger servers these days.  However, there is a nice
>Twisted tutorial based on this protocol, and it would be really nice if
>the examples worked more or less out of the box on Python 3,
>to give new users a nice experience, instead of running into a lot of
>errors.
>
>Based on your comments, I took a pass through the tutorial and changed
>the docstring for things like `IFingerSetterFactory` to bytes.  I also
>fixed up some of the examples in the finger tutorial.  Before, none of
>the examples worked on Python 3, but now quite a lot of them do.
>
>I didn't get to all of them, but if it is OK with you, I'd like to
>commit this stuff to begin with,
>and either I or other people can fix up the other examples over time.
>
>Thanks!
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/twisted/twisted/pull/689#issuecomment-275952841

-- 
Sent from my phone. Please excuse my brevity.

rodrigc commented:
@leth can you mark this review as approved on GitHub so I can list you as a reviewer in the commit message?

rodrigc commented:
Also, just out of curiousity, do you work on @einarfd 's team on this: https://github.com/twisted/twisted/pull/456#issuecomment-242096969

leth commented:
Not quite, we're in the same arm of the megacorp but different parts of it.
Our two groups have a history of collaboration; we have an internal project built on twisted which has an open-source-like collaboration model.

rodrigc committed:
Merge branch 'trunk' into 8230-rodrigc-finger

rodrigc commented:
@moshez thanks for writing this tutorial a long time ago.  I updates some of the examples to work in Python 3. :)
